### PR TITLE
feat(e2e): Automate creation of function app for E2E tests

### PR DESCRIPTION
**Pull request by vinagesh**
## Description of the changes
Automate creation of function app for E2E tests

## Reference/Link to the issue solved with this PR (if any)
https://msazure.visualstudio.com/One/_workitems/edit/6739275

### CODE REVIEW

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json [hunk 1]
```diff
@@ -51,6 +51,22 @@
             "metadata": {
                 "description": "The name of the DPS used by tests."
             }
+        },
+        "TestGetFunctionName": {
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json:55]: remove this, don't need a TestGet function for our tests

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json [hunk 2]
```diff
@@ -51,6 +51,22 @@
...
+            "type": "string"
+        },
+        "TestGetRunCsxContent": {
+            "type": "string"
+        },
+        "DpsCustomAllocatorFunctionName": {
+            "type": "string"
+        },
+
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json:64]: remove extra newline

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json [hunk 3]
```diff
@@ -51,6 +51,22 @@
             "metadata": {
                 "description": "The name of the DPS used by tests."
             }
+        },
+        "TestGetFunctionName": {
+            "type": "string"
+        },
+        "TestGetRunCsxContent": {
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json:58]: Should give these all defaults here

**Comment by timtay-microsoft** [comment]
Just as an aside, other SDK's still use the function app that I stood up for this SDK a while back. I like this change, but don't remove the existing function app without talking to the other SDKs first

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json (hunk 1 above)
Comment (vinagesh) [reply e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json:55]: I did not know that :) Will remove.

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json [hunk 4]
```diff
@@ -302,6 +318,113 @@
                 "recommendationsConfiguration": [],
                 "unmaskedIpLoggingStatus": "Enabled"
             }
+        },
+        {
+            "apiVersion": "2018-11-01",
+            "name": "[parameters('ResourceGroup')]",
```
[the remark this answers is not in the document]
Comment (vinagesh) [reply e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json:318]: Makes sense. I will resolve this in a PR and then you can make other changes :)

**Pull request merged**